linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/WasmContext.scala

```diff
@@ -177,6 +180,43 @@ final class WasmContext(
     )
   }
 
+  def genTypedClosureStructType(tpe0: ClosureType): (wanme.TypeID, wanme.TypeID) = {
+    // Normalize to the non-nullable variant
+    val tpe = tpe0.toNonNullable
+
+    typedClosureTypes.getOrElseUpdate(tpe, {
+      implicit val ctx = this
+
+      val funType = watpe.FunctionType(
+        watpe.RefType.struct :: tpe.paramTypes.map(TypeTransformer.transformParamType(_)),
+        TypeTransformer.transformResultType(tpe.resultType)
+      )
+      val funTypeID = genTypeID.forClosureFunType(tpe)
+      mainRecType.addSubType(funTypeID, OriginalName("fun" + tpe.show()), funType)
+
+      val fields: List[watpe.StructField] = List(
+        watpe.StructField(
+          genFieldID.typedClosure.data,
+          OriginalName("data"),
```

Comment:
Nothing to do here, but he use of `OriginalName` in the wasm subsystem throws me off quite often :-/

as in: why do we "materialize" an original name out of nowhere, what's the point, and then I remember: it's the only human readable name in the system. :shrug: